Replace social media widgets with social media links

radusuciu opened the pull request:
Removing social media widgets which are dynamically loaded has two main benefits:
- faster page loads
- better for privacy

The main difference is instead of clicking "Tweet" and being invited to tweet, you are linked to the Mixxx Twitter page (as requested in #13). In my opinion this is likely to increase engagement with Mixxx's social media. Would be nice if there was some data on how much use the widgets are getting.

The social media links in this PR are styled very similar to the widgets. Also removed jQuery as the only uses of Javascript on the website were simple enough to express without jQuery. It can always be added when needed.

radusuciu committed:
Actually include script in error page

Was not being included due to wrong block name

radusuciu committed:
Remove unnecessary css link fixing

radusuciu committed:
Replace use of jquery with very simple vanilla js

Note that we don't need to wait till document ready here because the extra script block is included at bottom of dom tree

radusuciu committed:
Move site javascript to own file

radusuciu committed:
Use vanilla js to box platform specific download

radusuciu committed:
Jquery not currently used

Can always be added back (maybe a more up to date version) if there starts to be a need for lots of DOM manipulation. As things stand now though, all uses of Javascript are trivial.

radusuciu committed:
Replace social widgets with links

radusuciu committed:
Move social links to separate template

radusuciu committed:
Re-use social links template in footer

radusuciu committed:
Adjust social icon styles for footer

radusuciu committed:
Add link to Mixxx FB page. Spacing adjustment for social links template

radusuciu committed:
Adjust wording re social media in footer

radusuciu committed:
Add license for ionic social media icons

radusuciu committed:
Add zulip to footer.

radusuciu committed:
Increase social icon font size slightly

radusuciu committed:
Add license for zulip logo

ronso0 commented:
I just now became really aware of the website repo..

I wanted to have a quick look at the icons (maybe apply Mixxx style?) but noticed the svgs are inline. does it have any benefit? do you mind storing the icons in separate files so they're easier to review and edit?

ronso0 commented:
okay, got it up & running.
before I never was aware of those buttons as my browser blocks fb items..
it's nice to get rid of tracking objects and just link to the social sites instead!

as I thought in the first place, we could omit the text and just use icons.
Without text the buttons would also be small enough to fit underneath **Manual**, next to **Press**. This would make the site look a bit tidier than now.

radusuciu commented:
Svg as files would provide a minor benefit through caching I suppose, but it may be balanced out by increased number of requests (unless on HTTP2) - I think. I don't recall exactly but I think I just did it because it was easier.. can of course convert to files

I can pick this back up at some point but I'm currently writing my PhD Thesis so... lol. I'll be done in May but feel free to take over if you want!

Also, I should mention that I was thinking of splitting this up into two pull requests since I also cleaned up some JS.

Be-ing commented:
I just remembered about this PR. Thank you for reducing external dependencies and getting rid of the tracking. I noticed that the icons can overlap the "DJ Your Way" text:
![image](https://user-images.githubusercontent.com/9455094/63603735-d2d9e000-c58f-11e9-9934-82cabef7a84c.png)
compared to the current website:
![image](https://user-images.githubusercontent.com/9455094/63603998-5b588080-c590-11e9-8ff5-c75a721917d9.png)

How about getting rid of the "Twitter" and "Facebook" text and just using their icons?

Be-ing commented:
Also thanks for adding the Zulip button :+1:

ronso0 commented:
yeah, that's pretty much what I proposed above and that would help in narrow view, too
> Without text the buttons would also be small enough to fit underneath Manual, next to Press. This would make the site look a bit tidier than now.

@radusuciu Are able to finish this or should someelse take over?

Be-ing commented on static/js/mixxx.js:
```diff
@@ -0,0 +1,32 @@
+
+function getOSName() {
+  var OSName="your OS";
+  if (navigator.appVersion.indexOf("Win")!=-1) OSName="Windows";
```
Let's use the same style we use for JS in controller mappings:
```suggestion
  if (navigator.appVersion.indexOf("Win") !== -1) {
    OSName="Windows";
  }
```

ronso0 commented:
yeah, that's pretty much what I proposed above and that would help in narrow view, too
> Without text the buttons would also be small enough to fit underneath Manual, next to Press. This would make the site look a bit tidier than now.

@radusuciu Are able to finish this or should someelse take over?

radusuciu commented:
@ronso0 @Be-ing I can take into account above suggestions and amend the PR!

radusuciu committed:
Create separate social sidebar

radusuciu committed:
Use svg files instead of hardcoded values

radusuciu committed:
Move social icon license to separate file.

radusuciu committed:
Indent social sidebar markup with 2 spaces to match project style.

radusuciu committed:
Simplify code for returning OS we're running on.

radusuciu committed:
Namespace js util functions.

radusuciu committed:
Extract download js to own file.

radusuciu committed:
Place social sidebar on all pages

I think it is unobstrusive enough that it merits inclusion on all pages. This could be especially applicable when users are linked to the download page directly for example.

radusuciu committed:
Revert "Place social sidebar on all pages"

This reverts commit 25673287c28e1e6471c744ead6dfb249065e3ce1.

radusuciu commented:
Reverted a commit that added these new side link icons to all pages, as some pages already had elements in the same position, and I think that that would warrant a separate PR.

radusuciu commented:
Other changes:
- moved inline svg to separate files
- re-organized js a bit. Namespaced what few functions there are and extracted some js from the download page to it's only file

ronso0 commented:
Looks good on desktop and mobile view, works as expected.
Are there any corner cases to test?

radusuciu commented:
> Looks good on desktop and mobile view, works as expected.
> Are there any corner cases to test?

Not that I can think of other than trying at different screen sizes.

Be-ing commented:
Thanks for taking care of this.
